### DISCUSSION TITLE
Min Max question.

### DISCUSSION THREAD

**Discussion by aegis72** [danthedeckie/simpleeval#135] [Q&A]
`max(0, min(X, X - 224)) if X >= 224 else X`

I have this equation/code that I think should ensure that, for given value X, the number provided will make the returned value either zero (if X would create a negative number when 224 is subtracted) allow the number to pass through unchanged if not, and would ensure that the number could not exceed the value of X minus 224.

My purpose is to ensure that a 224px square box can always fit within the confines of a larger box.

However, when I run this code in my context (a Stable Diffusion/ComfyUI node called "Evaluate Floats" that uses your library) I get the following error: 
```
Error occurred when executing Evaluate Floats:

Function 'max' not defined, for expression 'max(0, min(a, a - 224)) if a >= 224 else a'.


```
. Is my syntax simply wrongly formatted for your library?

**Comment by rysson** [comment]
Look into code, there is not `min` and `max` functions in `DEFAULT_FUNCTIONS`. 
You should set they directly
```python
SimpleEval(functions={'min': min,'max': max}, names={'a': 225}).eval('max(0, min(a, a - 224)) if a >= 224 else a')
```
